src/Catalyst.Core.Modules.Consensus.Tests/UnitTests/Deltas/DeltaVoterTests.cs

```diff
@@ -0,0 +1,365 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using Catalyst.Abstractions.Consensus.Deltas;
+using Catalyst.Abstractions.Hashing;
+using Catalyst.Abstractions.P2P;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.Util;
+using Catalyst.Core.Modules.Consensus.Deltas;
+using Catalyst.Core.Modules.Hashing;
+using Catalyst.Protocol.Peer;
+using Catalyst.Protocol.Wire;
+using Catalyst.TestUtils;
+using FluentAssertions;
+using LibP2P;
+using Microsoft.Extensions.Caching.Memory;
+using NSubstitute;
+using Serilog;
+using TheDotNetLeague.MultiFormats.MultiHash;
+using Xunit;
+
+namespace Catalyst.Core.Modules.Consensus.Tests.UnitTests.Deltas
+{
+    public sealed class DeltaVoterTests : IDisposable
+    {
+        public static readonly List<object[]> DodgyCandidates;
+
+        private readonly IHashProvider _hashProvider;
+        private readonly IMemoryCache _cache;
+        private readonly IDeltaProducersProvider _producersProvider;
+        private DeltaVoter _voter;
+        private readonly Cid _previousDeltaHash;
+        private readonly IList<PeerId> _producerIds;
+        private readonly PeerId _localIdentifier;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the field '_localIdentifier' and declare it as a local variable in the relevant methods.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)